### PR TITLE
updated to new api in Travis(v3)

### PR DESCRIPTION
**Pull request by bergpb**
Code updated into new Travis api v3, checked tests and some fixes applyed.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -34,6 +37,18 @@
                'Topic :: Software Development :: Quality Assurance',
                'Topic :: Software Development :: Testing']
 
+py_version = sys.version[:5]
```
Comment (sethmlarson) [inline setup.py]: Let's revert these changes. I'll always be using Python 3.6+ to deploy changes.

File: trytravis.py [hunk 2]
```diff
@@ -31,8 +31,8 @@
 __author__ = 'Seth Michael Larson'
 __email__ = 'sethmichaellarson@protonmail.com'
 __license__ = 'Apache-2.0'
-__url__ = 'https://github.com/SethMichaelLarson/trytravis'
-__version__ = '1.0.4'
+__version__ = '3'
```
Comment (sethmlarson) [inline trytravis.py]: Version should be `2.0` if we follow semver.

File: trytravis.py [hunk 3]
```diff
@@ -185,8 +218,29 @@ def _submit_changes_to_github_repo(path, url):
...
+        repos = json["repositories"]
+        for i in range(0, len(repos)):
+            if 'trytravis' in repos[i]["name"]:
+                id = repos[i]["id"]
+                return id
+
+
 def _wait_for_travis_build(url, commit, committed_at):
-    """ Waits for a Travis build to appear with the given commit SHA """
+    """ Waits for a Travis build to appear with the given commit SHA. """
```
Comment (sethmlarson) [inline trytravis.py]: Adding periods to doc-strings isn't related to this change, please revert.

File: trytravis.py [hunk 4]
```diff
@@ -320,9 +385,17 @@ def _slug_from_url(url):
...
 def _version_string():
     """ Gets the output for `trytravis --version`. """
+    """ Gets python version before get system information,
+    because methods like platorm.dist is not disponible
+    in python 3.8 and 3.9. """
+    py_version = sys.version[:5]
     platform_system = platform.system()
     if platform_system == 'Linux':
-        os_name, os_version, _ = platform.dist()
+        if '3.8' or '3.9' in py_version:
```
Comment (sethmlarson) [inline trytravis.py]: This change doesn't look related to the Travis v3 API. Could you revert it?

Also: This doesn't do what you think it does. This statement always evaluates to a truthy value.
I think you meant `'3.8' in py_version or '3.9' in py_version`?

Is there any reason in particular the previous implementation doesn't work?

File: trytravis.py [hunk 5]
```diff
@@ -335,22 +408,26 @@ def _version_string():
 
 def _travis_headers():
     """ Returns the headers that the Travis API expects from clients. """
-    return {'User-Agent': ('trytravis/%s (https://github.com/'
-                           'SethMichaelLarson/trytravis)') % __version__,
-            'Accept': 'application/vnd.travis-ci.2+json'}
+    token = _load_travis_token()
+    return {'Travis-API-Version': '%s' % __version__,
+            'User-Agent': 'API Explorer',
```
Comment (sethmlarson) [inline trytravis.py]: We shouldn't lie about our `User-Agent`. Please revert the user-agent change and hard-code the Travis-API-Version change.

**Review by sethmlarson — CHANGES_REQUESTED** [review]
There are a lot of changes here that aren't related to Travis v3 API.

File: setup.py (hunk 1 above)
Comment (bergpb) [reply setup.py]: okay but some builds like for python 2.7 maybe fail

File: trytravis.py (hunk 2 above)
Comment (bergpb) [reply trytravis.py]: ok, done

File: trytravis.py (hunk 3 above)
Comment (bergpb) [reply trytravis.py]: ok, done.

File: trytravis.py (hunk 4 above)
Comment (bergpb) [reply trytravis.py]: This project are builds in various python version, in version `3.8` and `3.9(nightly)` this library has no method dist(), and i verify python installed version and if `3.8` or `3.9(nightly)`, if true i use outher methods to get platform info. Without this build fails in `3.8` and `3.9(nightly)`.

[https://docs.python.org/3.8/library/platform.html](https://docs.python.org/3.8/library/platform.html)
[https://docs.python.org/3.9/library/platform.html](https://docs.python.org/3.9/library/platform.html)

Ps: Really, if condition return True ever, my mistake.

File: trytravis.py (hunk 5 above)
Comment (bergpb) [reply trytravis.py]: ok, done.

File: trytravis.py [hunk 6]
```diff
@@ -320,9 +385,17 @@ def _slug_from_url(url):
...
 def _version_string():
     """ Gets the output for `trytravis --version`. """
+    """ Gets python version before get system information,
+    because methods like platorm.dist is not disponible
+    in python 3.8 and 3.9. """
+    py_version = sys.version[:5]
     platform_system = platform.system()
     if platform_system == 'Linux':
-        os_name, os_version, _ = platform.dist()
+        if '3.8' in py_version or '3.9' in py_version:
```
Comment (sethmlarson) [inline trytravis.py]: Like I said in an earlier comment, we should be using `if hasattr(platform, "dist"):` then use `platform.dist()` otherwise fall back on `platform.system()` and `platform.version()`.

File: README.md [hunk 7]
```diff
@@ -183,18 +199,18 @@ supporters on our BountySource. Thank you for your support!
 This project is licensed under `Apache-2.0`. Read [`LICENSE`](https://github.com/SethMichaelLarson/trytravis/blob/master/LICENSE) for more information.
 
 ```
- Copyright 2017 Seth Michael Larson
+Copyright 2017 Seth Michael Larson
```
Comment (sethmlarson) [inline README.md]: This change is nice but not necessary for Travis v3 API

File: .travis/install.sh [hunk 8]
```diff
@@ -11,29 +11,25 @@ if [[ "$(uname -s)" == 'Darwin' ]]; then
...
-            pyenv global 3.3.6
-            ;;
-        py34)
-            pyenv install 3.4.5
-            pyenv global 3.4.5
-            ;;
         py35)
-            pyenv install 3.5.2
-            pyenv global 3.5.2
+            pyenv install 3.5.9
```
Comment (sethmlarson) [inline .travis/install.sh]: This change is nice but not necessary for Travis v3 API.

File: setup.py [hunk 9]
```diff
@@ -34,6 +35,10 @@
                'Topic :: Software Development :: Quality Assurance',
                'Topic :: Software Development :: Testing']
 
+file_dir = os.path.join(base_dir, 'CHANGELOG.md')
```
Comment (sethmlarson) [inline setup.py]: This change is nice but not necessary for Travis v3 API. Can be done in a separate PR.

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Still needs changes to make this PR a minimal change to support Travis v3.

File: trytravis.py (hunk 6 above)
Comment (bergpb) [reply trytravis.py]: ok, i changed

File: README.md (hunk 7 above)
Comment (bergpb) [reply README.md]: ok, reverted to original file and added instructions to use with v3 API

File: .travis/install.sh (hunk 8 above)
Comment (bergpb) [reply .travis/install.sh]: okay, reverted to original file.

File: setup.py (hunk 9 above)
Comment (bergpb) [reply setup.py]: okay reverted to original file.
